src/main/java/org/apache/brooklyn/cloudfoundry/location/CloudFoundryClientRegistry.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.cloudfoundry.location;
+
+
+import org.apache.brooklyn.util.core.config.ConfigBag;
+
+public interface CloudFoundryClientRegistry {
+
+    CloudFoundryPaasClient getCloudFoundryClient(ConfigBag conf, boolean allowReuse);
```

Comment:
Anything, this PR is only the first step to delete avoid the `CloudFoundryPaasClient`.  A fake client was added, and the tests were modified. The next step is move all the code from `CloudFoundryPaasClient` to `CloudFoundryPaasLocation.

(I found some mistakes in the `PaasCloudFoundryLiveTests`,  they are  fixed in this RP)